sdk/eventgrid/azure-eventgrid/azure/eventgrid/aio/_helpers_async.py

```diff
@@ -0,0 +1,29 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+from azure.core.pipeline.policies import AzureKeyCredentialPolicy, AsyncBearerTokenCredentialPolicy
+from azure.core.credentials import AzureKeyCredential, AzureSasCredential
+
+from .. import _constants as constants
+from .._signature_credential_policy import EventGridSasCredentialPolicy
+
+def _get_authentication_policy_async(credential):
+    if credential is None:
+        raise ValueError("Parameter 'self._credential' must not be None.")
+    if hasattr(credential, "get_token"):
```

Comment:
dumb question: what error would be raised if a user passed a sync AD credential into the async client?